src/grid/interpolate.jl

```diff
@@ -864,4 +864,23 @@ function differentiate1D(::CompositeDifferentiate, data, xgrid, x)
     return differentiate1D(view(data, head:tail), xgrid.subgrids[i], x)
 end
 
+# locate and volumn for monte carlo
+
+function locate(grid::AbstractGrid, x)
+    @assert x >= grid.bound[1] && x <= grid.bound[2]
+    i = floor(grid, x)
+    return abs(x-grid[i])<abs(x-grid[i+1]) ? i : (i+1)
+end
+
```

Comment:
documentation. Something like: 

returns the span of the grid point. If the grid points are ...x_{i-1}, x_i, x_{i+1},..., then the volume of x_i is defined as (x_{i+1}-x_{i-1})/2. The volume of the grid points on the boundary is halved.